### PR TITLE
React Migration

### PR DESCRIPTION
**Pull request by RayHCai**

### CODE REVIEW

File: question-generation/reflectia/assets/logo-filled.png
Comment (nghtctrl) [inline question-generation/reflectia/assets/logo-filled.png:1]: Should we leave these assets alone for future development? Front-end developers and UI researchers might be interested.

**Comment by nghtctrl** [comment]
Looks good to me! This is something that I brought up with @kcarnold some time ago, but would we benefit from creating a separate repository named `question-generation` (or another appropriate name) instead of bothering about moving old things around into separate folder for each summer research sessions? This repository now looks like a small version of GitHub Teams rather than an actual repository.

**Review by nghtctrl — APPROVED** [review]

File: question-generation/reflectia/assets/logo-filled.png
Comment (RayHCai) [reply question-generation/reflectia/assets/logo-filled.png:1]: They are just default images that say Logo. I added one of them back as it seems like the asset folder cannot be empty.

**Comment by nghtctrl** [comment]
> Looks good to me! This is something that I brought up with @kcarnold some time ago, but would we benefit from creating a separate repository named `question-generation` (or another appropriate name) instead of bothering about moving old things around into separate folder for each summer research sessions? This repository now looks like a small version of GitHub Teams rather than an actual repository.

This is not too important though. I realize that our GitHub Teams keeps three repositories, each signifying different desirable components of our writing system.

**Review by nghtctrl — APPROVED** [review]
Looks good to me!

**Comment by RayHCai** [comment]
I think it would be beneficial if we moved the old research code to a different org or repo

**Comment by nghtctrl** [comment]
> I think it would be beneficial if we moved the old research code to a different org or repo

Another small benefit is that, future researchers won't accidentally PR to high-level forks, which might cause unnecessary confusion.

**Comment by kcarnold** [comment]
I agree that the organization of this repo is not good, and that the fact that GitHub considers it a fork has been sometimes detrimental.

But I think it's helpful that we're in a single repo for frontend and backend, though, since many changes will affect both. And I think our custom editor frontend should probably be a peer to the Word addin code, reusing API where that makes sense.

So perhaps we go to:

```
- backend/
  - server.py
  - nlp.py
  - nlp_tests.py
- addin/
  - manifest.xml
  - package.json
  - src/
  - etc.
- custom_editor/ (or whatever we call it)
  - ...
```

but *not in this PR*. (Can someone move this to a separate Issue?)

**Comment by kcarnold** [comment]
Now I'm looking at the code. **Each PR should do one thing.** Reorganizing the repo is a separate thing. Resist the urge to get it all done at once; that can slow down merging one part if there's discussion on another part.

File: question-generation/reflectia/src/components/app.tsx
Comment (kcarnold) [inline question-generation/reflectia/src/components/app.tsx:95]: How about we just let this be the currently selected preset prompt?

File: question-generation/reflectia/src/components/app.tsx
Comment (kcarnold) [inline question-generation/reflectia/src/components/app.tsx:44]: There's a fancy React thing called "Suspense" that makes a slick way to do this... but this is *totally fine*.

File: question-generation/server.py
Comment (kcarnold) [inline question-generation/server.py:80]: Is this an exact repeat of the above?

**Review by kcarnold — APPROVED** [review]
Seems like a very sensible and pragmatic React conversion, let's land it.

There's some other unrelated changes in here. I reverted the most distracting of them since we're still discussing that. In general, the "Files Changed" tab in a PR should make sense.

I left some comments that really belong as TODOs in the code, not to block this PR merging. I'm going to hit Merge, but @RayHCai or @nghtctrl  please go back through the comments and add code comments accordingly (or just fix if it's easy)

**Comment by kcarnold** [comment]
I just saw that @SophiaSun18  was also requested for review. Her comments and visibility here would also be helpful. I'm going to merge just so we're not blocked, but Sophia you can still leave comments and questions here and the authors should deal with them (possibly in a future PR)

File: question-generation/reflectia/src/taskpane.css
Comment (SophiaSun18) [inline question-generation/reflectia/src/taskpane.css:33]: This .cards-container seems to be repeating with the .card-container below? Maybe remove this?

File: question-generation/reflectia/src/components/app.tsx
Comment (SophiaSun18) [reply question-generation/reflectia/src/components/app.tsx:64]: I'll check out if there's a method to go though the whole document and record all the highlighted text before the add-in start doing the paragraph highlighting.

Comment (kcarnold) [reply question-generation/reflectia/src/components/app.tsx:64]: Recording all previous highlights seems like it might get complicated. If there's not an easy way to do this, let's just table it for now. (It'll be easy in our custom editor.)

**Comment by RayHCai** [comment]
Should I create a separate PR to move the 2022 files? I think its a little distracting having them in the root directory

**Comment by nghtctrl** [comment]
@RayHCai I would suggest creating a new repo named `reflections` and migrating our code there, following [this structure](https://github.com/AIToolsLab/questions/pull/18#issuecomment-1635892725) suggested by @kcarnold. Creating a new PR for that would be helpful.

**Pull request merged**
